activerecord/lib/active_record/base.rb

```diff
@@ -2181,11 +2181,16 @@ def with_scope(method_scoping = {}, action = :merge, &block)
             end
           end
 
+          initial_scoped_methods = self.scoped_methods.dup
           self.scoped_methods << method_scoping
           begin
-            yield
+            begin
+              yield
+            ensure
+              self.scoped_methods.replace(initial_scoped_methods)
```

Comment:
imo - now that we're changing this line, we might as well remove `self` - opinions @outoftime ?